metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/asa/BasicAsaParser.java

```diff
@@ -0,0 +1,165 @@
+        String syslogPattern = "%{CISCO_TAGGED_SYSLOG}";
+        JSONObject metronJson = new JSONObject();
+        List<JSONObject> messages = new ArrayList<>();
+        try {
+            String logLine = new String(rawMessage, "UTF-8");
+            LOG.trace("[Metron] Started parsing raw message: " + logLine);
+
+            asaGrok.compile(syslogPattern);
+            Match syslogMatch = asaGrok.match(logLine);
+            syslogMatch.captures();
+            Map<String, Object> syslogJson = syslogMatch.toMap();
+            LOG.trace("[Metron] Grok CISCO syslog matches: " + syslogMatch.toJson());
+
+            metronJson.put(Constants.Fields.ORIGINAL.getName(), logLine);
+            metronJson.put(Constants.Fields.TIMESTAMP.getName(),
+                    SyslogUtils.parseTimestampToEpochMillis((String) syslogJson.get("CISCOTIMESTAMP")));
+            metronJson.put("ciscotag", syslogJson.get("CISCOTAG"));
+            metronJson.put("syslog_severity", SyslogUtils.getSeverityFromPriority((int) syslogJson.get("syslog_pri")));
+            metronJson.put("syslog_facility", SyslogUtils.getFacilityFromPriority((int) syslogJson.get("syslog_pri")));
+
+            String messagePattern = patternMap.get(syslogJson.get("CISCOTAG"));
+            asaGrok.compile("%{"+ messagePattern +"}");
+            Match messageMatch = asaGrok.match((String) syslogJson.get("message"));
+            messageMatch.captures();
+            Map<String, Object> messageJson = messageMatch.toMap();
+            LOG.trace("[Metron] Grok CISCO message matches: " + messageMatch.toJson());
+
+            String src_ip = (String) messageJson.get("src_ip");
+            if (src_ip != null && ipValidator.isValid(src_ip))
+                metronJson.put(Constants.Fields.SRC_ADDR.getName(), src_ip);
+
+            Integer src_port = (Integer) messageJson.get("src_port");
+            if (src_port != null && FieldValidators.isValidPort(src_port))
+                metronJson.put(Constants.Fields.SRC_PORT.getName(), src_port);
+
+            String dst_ip = (String) messageJson.get("dst_ip");
+            if (dst_ip != null && ipValidator.isValid(dst_ip))
+                metronJson.put(Constants.Fields.DST_ADDR.getName(), dst_ip);
+
+            Integer dst_port = (Integer) messageJson.get("dst_port");
+            if (dst_port != null && FieldValidators.isValidPort(dst_port))
+                metronJson.put(Constants.Fields.DST_PORT.getName(), dst_port);
+
+            String protocol = (String) messageJson.get("protocol");
+            if (protocol != null)
+                metronJson.put(Constants.Fields.PROTOCOL.getName(), protocol.toLowerCase());
+
+            String action = (String) messageJson.get("action");
+            if (action != null)
+                metronJson.put("action", action.toLowerCase());
+
+            LOG.trace("[Metron] Final normalized message: " + metronJson.toString());
+
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Probably better to do a LOG.error and grab the e.getMessage() along with other contextual information. Need to make it easy to understand why it blew up.